### ISSUE TITLE
Port over non-redundant Span tests from corefxlab

### ISSUE THREAD

**Issue by ahsonkhan** [dotnet/corefx#28281]
From https://github.com/dotnet/corefxlab/issues/1403

Add span-based tests from corefxlab that are non-redundant and are testing interesting scenarios.

From a cursory view, here are some examples:
https://github.com/dotnet/corefxlab/blob/5b2bbc602833e4fac9a2cfe05ddfafe612dd9e86/tests/System.Buffers.Primitives.Tests/BasicUnitTests.cs#L332
https://github.com/dotnet/corefxlab/blob/5b2bbc602833e4fac9a2cfe05ddfafe612dd9e86/tests/System.Buffers.Primitives.Tests/CastTests.cs#L54
https://github.com/dotnet/corefxlab/blob/5b2bbc602833e4fac9a2cfe05ddfafe612dd9e86/tests/System.Buffers.Primitives.Tests/CastTests.cs#L90
https://github.com/dotnet/corefxlab/blob/5b2bbc602833e4fac9a2cfe05ddfafe612dd9e86/tests/System.Buffers.Primitives.Tests/CastTests.cs#L126
https://github.com/dotnet/corefxlab/blob/5b2bbc602833e4fac9a2cfe05ddfafe612dd9e86/tests/System.Buffers.Primitives.Tests/OverflowTests.cs#L27

**Comment by sputier** [comment]
I'd like to help on this one 😄

**Comment by ahsonkhan** [comment]
Great. Go for it :)

**Comment by sputier** [comment]
Sorry, I didn't have much time to work on it... Family first ! I'll work on it this week. Thanks for the reminder !

**Comment by WinCPP** [comment]
Hi @ahsonkhan is this still up for grabs? Also does this go int src/System.Memory/tests/Span project?

Thanks,
Mandar

**Comment by ahsonkhan** [comment]
>  is this still up for grabs

Yes it is. We still have some leftover tests out side of the commit.

> Also does this go into src/System.Memory/tests/Span project?

Yes.

**Comment by WinCPP** [comment]
Looking into these!

**Comment by igor-susic** [comment]
@ahsonkhan I guess no one ported these tests yet if this one is still available I would like to wrap it up :smile:

**Comment by igor-susic** [comment]
@ahsonkhan Just a little update, as I was on tight personal schedule, I did not made great progress yet, but expect PR soon. Also, do you have any convenient method to monitor what tests already exists and what should I port and what I have already ported, I created some kind of "todo" list but, I find my self spending little bit more time than expected looking these things up :smile:  

Ps. Should I share my todo list here if it contains all `Span` tests from `corefxlab` ?
